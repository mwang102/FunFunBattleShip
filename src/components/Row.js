import React from 'react'
import Squares from './Square'
import styled, { keyframes } from "styled-components";
import PropTypes from 'prop-types'


export default class Row extends React.Component{
  static propTypes = {
    handlePick:PropTypes.func.isRequired,
    rowNumber:PropTypes.number.isRequired
  }

  render(){
    let squares = this.props.data.map((square,index)=>{

        return <Squares handlePick={this.props.handlePick} 
                rowNumber={this.props.rowNumber} 
                columnNumber={index} key={index} data={square}/>
    })

    return(
      <RowContainer rowNumber={this.props.rowNumber}>
        {squares}
      </RowContainer>
    )
  }
}

const visibility = (delay) => {
  return keyframes`
  from,
  60%,
  75%,
  90%,
  to {
    -webkit-animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
  }

  from {
    opacity: 0;
    -webkit-transform: translate3d(0, 3000px, 0);
    transform: translate3d(0, 3000px, 0);
  }

  60% {
    opacity: 1;
    -webkit-transform: translate3d(0, -20px, 0);
    transform: translate3d(0, -20px, 0);
  }

  75% {
    -webkit-transform: translate3d(0, 10px, 0);
    transform: translate3d(0, 10px, 0);
  }

  90% {

    -webkit-transform: translate3d(0, -5px, 0);
    transform: translate3d(0, -5px, 0);
  }

  to {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }

`
};

const RowContainer = styled.div`
    display:flex;
    flex: 1 0 auto;
    margin:5px;
    background:ivory;
    height:auto;
    animation: ${props => visibility(props.rowNumber)}};
    animation-duration: 1s;
    animation-iteration-count: 1;
    animation-timing-function: ease;
    animation-fill-mode: both;
    animation-delay:${props => '.' + props.rowNumber + 's'
    }};
`;
