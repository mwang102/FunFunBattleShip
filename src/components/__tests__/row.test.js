import React from 'react';
import { shallow } from 'enzyme'
import Row from '../Row'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('row component',()=>{
	it('check for number of squares in each row', ()=>{
		const mockRow = [[5,false],[0,false],[0,false],[0,false]]

		const mockCallback = jest.fn()

		const wrapper = shallow(<Row data={mockRow} handlePick={mockCallback} 
									 rowNumber={1} columnNumber={1}/>)

		const squares = wrapper.find('Square')
		expect(squares.length).toEqual(4)
	})

	it('matches the snapshot', ()=>{
		const mockRow = [[5,false],[0,false],[0,false],[0,false]]

		const mockCallback = jest.fn()
		const tree = renderer.create(<Row data={mockRow} handlePick={mockCallback} 
									 rowNumber={1} columnNumber={1}/>).toJSON()
		
		expect(tree).toMatchSnapshot()
	})
})