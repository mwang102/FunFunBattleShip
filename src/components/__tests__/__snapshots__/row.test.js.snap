// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row component matches the snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px 5px;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 0px 1.0875rem 1.45rem;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.5);
  font-size: 30px;
  background-repeat: no-repeat;
  background-position: center;
  background: BlanchedAlmond;
  cursor: pointer;
}

.c1:hover {
  background: gold;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 5px;
  background: ivory;
  height: auto;
  -webkit-animation: cJUYLg;
  animation: cJUYLg;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: .1s;
  animation-delay: .1s;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  />
  <div
    className="c1"
    onClick={[Function]}
  />
  <div
    className="c1"
    onClick={[Function]}
  />
  <div
    className="c1"
    onClick={[Function]}
  />
</div>
`;
